### PR TITLE
Weather lookup bslib simplified

### PR DESCRIPTION
**Pull request by nstrayer**
Adds a light and dark mode toggle plus the use of `bs_add_rules()` and some `scss` code to make custom elements match the current theme. For the new `bslib` post on shiny dev center

### CODE REVIEW

**Comment by cpsievert** [comment]
Shoot, looks like thematic doesn't influence this `textGrob()`, but you can fix it by adding `gp = grid::gpar(col = thematic::thematic_get_option("fg"))` to `textGrob()`

https://github.com/rstudio/shiny_app_stories/pull/2/files#diff-d6f607036893ff8d0f3fc4142cdd8fb77f019c24ebd31d6437a9379a91aab1adR161


Screen Shot 2021-03-05 at 3 29 03 PM

File: weather_lookup/app.R [hunk 1]
```diff
@@ -200,7 +190,14 @@ server <- function(input, output, session) {
               panel.grid.minor = element_line(color = "grey85", size = 0.2))
```
Comment (cpsievert) [inline weather_lookup/app.R:193]: Just curious, why is this here?

File: weather_lookup/app.R [hunk 2]
```diff
@@ -6,32 +6,17 @@ library(patchwork)
...
-  print("Enabling caching")
-} else {
-  bindCache <- function(x, ...) x
-  print("Disabling caching")
-}
 
 source('helpers.R')
-
+options(shiny.autoreload = TRUE)
+shiny::devmode(TRUE)
```
Comment (cpsievert) [inline weather_lookup/app.R:12]: Do you really want to include this?

**Review by cpsievert — APPROVED** [review]
Great work, LGTM once comments are addressed

File: weather_lookup/app.R (hunk 1 above)
Comment (nstrayer) [reply weather_lookup/app.R:193]: I had thought that the gray lines were less confusing given the temperature line is white itself. However, this was before the dark-mode was added so I'm less certain now

File: weather_lookup/app.R (hunk 2 above)
Comment (nstrayer) [reply weather_lookup/app.R:12]: I do not!

File: weather_lookup/app.R (hunk 1 above)
Comment (nstrayer) [reply weather_lookup/app.R:193]: Either way a comment should have been there to explain the "why", added one.
